src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackEditPageDataTest.java

```diff
@@ -0,0 +1,330 @@
+        
+        newQuestionForm = data.getNewQnForm();
+        assertEquals(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                        .withUserId(instructor.googleId)
+                        .withCourseId(fs.courseId)
+                        .withSessionName(fs.feedbackSessionName).toString(), newQuestionForm.getDoneEditingLink().toString());
+        assertFalse(newQuestionForm.getFeedbackPathSettings().isNumberOfEntitiesToGiveFeedbackToChecked());
+        
+        assertEquals(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                         .withUserId(instructor.googleId)
+                         .withCourseId(fs.courseId)
+                         .withSessionName(fs.feedbackSessionName).toString(),
+                     newQuestionForm.getDoneEditingLink().toString());
+        
+        
+        copyForm = data.getCopyQnForm();
+        assertEquals(0, copyForm.getQuestionRows().size());
+        
+        
+        ______TS("instructor with insufficient permissions");
+        // setup
+        data = new InstructorFeedbackEditPageData(dataBundle.accounts.get("helperOfCourse1"));
+        fs = dataBundle.feedbackSessions.get("session1InCourse1");
+        fs.resultsVisibleFromTime = Const.TIME_REPRESENTS_FOLLOW_VISIBLE;
+        fs.sessionVisibleFromTime = Const.TIME_REPRESENTS_FOLLOW_OPENING;
+        
+        questions = new ArrayList<FeedbackQuestionAttributes>();
+        questions.add(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1"));
+        questions.add(dataBundle.feedbackQuestions.get("qn2InSession1InCourse1"));
+        
+        copiableQuestions = new ArrayList<FeedbackQuestionAttributes>();
+        copiableQuestions.addAll(dataBundle.feedbackQuestions.values());
+        
+        questionHasResponses = new HashMap<String, Boolean>();
+        questionHasResponses.put(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1").getId(), true);
+        
+        studentList = new ArrayList<StudentAttributes>();
+        studentList.add(dataBundle.students.get("student1InCourse1"));
+        
+        instructorList = new ArrayList<InstructorAttributes>();
+        instructorList.add(dataBundle.instructors.get("instructor1OfCourse1"));
+        
+        instructor = getInstructorFromBundle("helperOfCourse1");
+        
+        data.init(fs, questions, copiableQuestions, questionHasResponses, studentList, instructorList, instructor);
+        assertEquals(0, data.getCopyQnForm().getQuestionRows().size());
+    }
+    
+    public InstructorAttributes getInstructorFromBundle(String instructor) {
+        InstructorAttributes instructorAttributes = dataBundle.instructors.get(instructor);
+        instructorAttributes.privileges = gson.fromJson(instructorAttributes.instructorPrivilegesAsText, InstructorPrivileges.class);
+        return instructorAttributes;
+    }
+
+    public void verifyMapContains(Map<String, Boolean> map, List<FeedbackParticipantType> list) {
```

Comment:
You can eliminate the `Arrays.asList()`s in testAll by using the `...` varargs construct for `FeedbackParticipantType` here for shorter code.
It's fine as it is right now too, though, with a clear distinction between the map and participant types.
Another alternative would be an explicit normal array (`[]`) that uses `{...}` instead of `Arrays.asList(...)`.